src/Context/TagsContext.php

```diff
@@ -21,52 +21,53 @@ public function merge(array $data, bool $recursive = false): void
             throw new \InvalidArgumentException('The tags context does not allow recursive merging of its data.');
         }
 
-        foreach ($data as $value) {
-            if (!\is_string($value)) {
-                throw new \InvalidArgumentException('The $data argument must contains a simple array of string values.');
-            }
-        }
-
-        parent::merge($data);
+        parent::merge(self::sanitizeTags($data));
     }
 
     /**
      * {@inheritdoc}
      */
     public function setData(array $data): void
     {
-        foreach ($data as $value) {
-            if (!\is_string($value)) {
-                throw new \InvalidArgumentException('The $data argument must contains a simple array of string values.');
-            }
-        }
-
-        parent::setData($data);
+        parent::setData(self::sanitizeTags($data));
     }
 
     /**
      * {@inheritdoc}
      */
     public function replaceData(array $data): void
     {
-        foreach ($data as $value) {
-            if (!\is_string($value)) {
-                throw new \InvalidArgumentException('The $data argument must contains a simple array of string values.');
-            }
-        }
-
-        parent::replaceData($data);
+        parent::replaceData(self::sanitizeTags($data));
     }
 
     /**
      * {@inheritdoc}
      */
     public function offsetSet($offset, $value): void
     {
+        if (is_numeric($value)) {
+            $value = (string) $value;
+        }
+
         if (!\is_string($value)) {
             throw new \InvalidArgumentException('The $value argument must be a string.');
         }
 
         parent::offsetSet($offset, $value);
     }
+
+    private static function sanitizeTags(array $tags): array
```

Comment:
Missing the description for both the function and the parameter. Also, maybe to be consistent with the names of the rest of the methods we could call this function `sanitizeData`, the parameter `$data` and the value being checked `$item`